kanidmd/src/lib/credential/mod.rs

```diff
@@ -706,6 +768,59 @@ impl Credential {
         }
     }
 
+    pub(crate) fn update_backup_code(&self, backup_code: HashSet<String>) -> Self {
+        let type_ = match &self.type_ {
+            // TODO: we should panic/reject CredentialType != PasswordMfa?
```

Comment:
we should reject (not panic), so this would change to `Result<Self, OperationError>` to represent that errorstate